src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Browser.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+
+using System.ComponentModel;
+using System.Collections.Generic;
+using System.Net.Sockets;
+using System.Runtime.Versioning;
+using System.Threading;
+
+namespace System.Net.NetworkInformation
+{
+    public partial class NetworkChange
+    {
+        private static bool s_isBrowserNetworkChangeListenerAttached;
+        private static event NetworkAvailabilityChangedEventHandler? s_networkAvailabilityChanged;
+
+        [UnsupportedOSPlatform("illumos")]
+        [UnsupportedOSPlatform("solaris")]
+        public static event NetworkAvailabilityChangedEventHandler? NetworkAvailabilityChanged
+        {
+            add
+            {
+                if (!s_isBrowserNetworkChangeListenerAttached)
```

Comment:
This should happen on the main thread, but don't have a way to check for it yet

cc @lambdageek

Reply:
It would be good to add some guard which will fail on other threads, until we know how to dispatch it onto main thread. 
Same would be good for other things like WS or HTTP.

Reply:
I agree, but according to discussion with Aleksey, we don't have anything yet available in C# and probably this PR is not a place to add it.